app/src/main/java/com/mstudio/superheroarch/GetCharacterDetailsUseCase.kt

```diff
@@ -0,0 +1,23 @@
+package com.mstudio.superheroarch
+
+class GetCharacterDetailsUseCase {
+
+    private val repository = RickAndMortyRepository()
+    private val tmdbRepository = TMDBRepository()
+
+    suspend fun getCharacterDetails(character: Character): CharacterUIEntity? {
```

Comment:
What about returning a Character instead of a UIEntity? IMO, the UseCase should not be tied to the UI.